### PR TITLE
Add a skeleton for the webhooklistener code.

### PR DESCRIPTION
**Pull request by schmidtw**

### CODE REVIEW

File: webhooklisteners/webhooksharedlist.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package webhooklisteners
+
+import (
+//"time"
+)
+
+// WebHookSharedListFactory provides the input values to create a
+// WebHookSharedList service object with New()
+type WebHookSharedListFactory struct {
```
Comment (johnabass) [inline webhooklisteners/webhooksharedlist.go]: Why have a factory?  It seems like all you need is a simple list type.

File: webhooklisteners/webhooksharedlist.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+// Add adds a WebHookListener to the shared slist
+func (w *WebHookSharedList) Add(item WebHookListener) (err error) {
+	// TODO the work of sending this to AWS
+	return
+}
+
+// Listen adds a listener function to be notifed when there is a change in the
+// list.  The entire list is given to the listener for processing.
+func (w *WebHookSharedList) Listen(func([]WebHookListener)) (err error) {
```
Comment (johnabass) [inline webhooklisteners/webhooksharedlist.go]: Is this intended to be some sort of goroutine?  If so, I wouldn't recommend it.  A slice guarded by a sync.RWMutex is faster, better, and easier to maintain.

File: webhooklisteners/webhooksharedlist.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// New creates a new WebHookSharedList service object
+func (w *WebHookSharedListFactory) New() (out *WebHookSharedList, err error) {
+	// TODO do the work
+	return
+}
+
+// WebHookSharedList is the interface that provides the shared list of webhooks
+// across the system
+type WebHookSharedList struct {
+	// TODO add parameters needed to operate the service
```
Comment (johnabass) [inline webhooklisteners/webhooksharedlist.go]: If all you need is a simple slice, consider just having this as `type WebHookSharedList []WebHookListener`

**Comment by schmidtw** [comment]
Do not want.

**Pull request closed without merging**